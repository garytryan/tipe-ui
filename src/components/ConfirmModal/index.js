export { default } from './ConfirmModal.vue'
