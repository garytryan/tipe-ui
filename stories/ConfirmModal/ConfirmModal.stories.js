import { storiesOf } from '@storybook/vue'

import ConfirmModal from '../../src/components/ConfirmModal'

const style = () => ({
  width: '100vw',
  height: '100vh',
  'background-color': 'rgba(0, 0, 0, 0.3)',
  display: 'flex',
  'align-items': 'center',
  'justify-content': 'center'
})

storiesOf('Confirm Modal', module).add('confirm modal', () => ({
  components: { ConfirmModal },
  computed: { style },
  template:
    '<div :style="style"><confirm-modal title="Delete Member?" subtitle="Are you sure you want to delete this member from the member list?" buttonText="DELETE"/></div>'
}))
