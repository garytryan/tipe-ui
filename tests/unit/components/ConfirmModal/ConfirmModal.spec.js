import { shallowMount } from '@vue/test-utils'
import ConfirmModal from '@/components/ConfirmModal'
import { createRenderer } from 'vue-server-renderer'

import TipeIcon from '@/components/Icon'
import TipeButton from '@/components/Button'
import TipeSeperator from '@/components/Seperator'

const props = {
  title: 'I am the title',
  subtitle: 'I am the subtitle',
  buttonText: 'Delete'
}
describe('AddMember.vue', () => {
  it('renders', () => {
    const renderer = createRenderer()
    const wrapper = shallowMount(ConfirmModal, {
      propsData: {
        ...props
      }
    })
    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) throw new Error(err)
      expect(str).toMatchSnapshot()
    })
  })
  it('has correct data-tipe-ui attibute', () => {
    const wrapper = shallowMount(ConfirmModal, {
      propsData: {
        ...props
      }
    })
    expect(wrapper.attributes()['data-tipe-ui']).toBe('TipeConfirmModal')
  })
  it('should render the TipeIcon component', () => {
    const wrapper = shallowMount(ConfirmModal, {
      propsData: {
        ...props
      }
    })
    expect(wrapper.contains(TipeIcon)).toBe(true)
  })
  it('should render the TipeButton component', () => {
    const wrapper = shallowMount(ConfirmModal, {
      propsData: {
        ...props
      }
    })
    const buttons = wrapper.findAll(TipeButton)
    expect(buttons).toHaveLength(2)
  })
  it('should render the TipeSeperator component', () => {
    const wrapper = shallowMount(ConfirmModal, {
      propsData: {
        ...props
      }
    })
    expect(wrapper.contains(TipeSeperator)).toBe(true)
  })
  it('should have the grid class', () => {
    const wrapper = shallowMount(ConfirmModal, {
      propsData: {
        ...props
      }
    })
    expect(wrapper.classes()).toContain('grid')
  })
  it('should have the correct props', () => {
    const wrapper = shallowMount(ConfirmModal, {
      propsData: {
        ...props
      }
    })
    expect(wrapper.props().title).toBe(props.title)
    expect(wrapper.props().subtitle).toBe(props.subtitle)
    expect(wrapper.props().buttonText).toBe(props.buttonText)
  })
  it('should emit the close method', () => {
    const wrapper = shallowMount(ConfirmModal, {
      propsData: {
        ...props
      }
    })
    wrapper.vm.$emit('close')
    expect(wrapper.emitted().close).toBeTruthy()
  })
  it('should emit the action method', () => {
    const wrapper = shallowMount(ConfirmModal, {
      propsData: {
        ...props
      }
    })
    wrapper.vm.$emit('action')
    expect(wrapper.emitted().action).toBeTruthy()
  })
  it('should emit the cancel method', () => {
    const wrapper = shallowMount(ConfirmModal, {
      propsData: {
        ...props
      }
    })
    wrapper.vm.$emit('cancel')
    expect(wrapper.emitted().cancel).toBeTruthy()
  })
})
