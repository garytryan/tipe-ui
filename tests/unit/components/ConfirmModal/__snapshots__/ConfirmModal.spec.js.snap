// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddMember.vue renders 1`] = `
<div data-tipe-ui="TipeConfirmModal" class="grid">
  <div class="close">
    <TipeIcon-stub></TipeIcon-stub>
  </div>
  <div class="content">
    <p class="title">I am the title</p>
    <p class="subtitle">I am the subtitle</p>
    <div class="seperator">
      <TipeSeperator-stub></TipeSeperator-stub>
    </div>
    <TipeButton-stub class="cancelBtn"></TipeButton-stub>
    <TipeButton-stub class="rightBtn"></TipeButton-stub>
  </div>
</div>
`;
